### ISSUE TITLE
iOS Navigate SDK: RoutingError::INTERNAL_ERROR

### ISSUE THREAD

**Issue by LaurentiuUngur** [heremaps/here-sdk-examples#275]
Hi. When I'm trying to calculate same route in same app instance, the SDK returns following error:

<img width="687" alt="Screenshot 2023-10-02 at 14 17 00" src="https://github.com/heremaps/here-sdk-examples/assets/13499785/98faa04c-bbe8-4a56-becb-a8244375e42e">

1st calculation of a route from Berlin to Budapest:
<img width="504" alt="Screenshot 2023-10-02 at 14 16 29" src="https://github.com/heremaps/here-sdk-examples/assets/13499785/4f83c957-8fa3-4cd7-a355-b273b10d6f67">

2nd calculation of a route from Berlin to Budapest:
<img width="687" alt="Screenshot 2023-10-02 at 14 17 00" src="https://github.com/heremaps/here-sdk-examples/assets/13499785/ea2a5e54-37ac-44ae-a6cd-525888ee522d">

If I force close the app, re-run, I'm able to calculate again same route.

Tested with SDK version 4.15.3, 4.15.4, 4.15.5 (latest), I haven't checked on older versions.

Any idea why it's this happening ? On HereSDK Premium Edition this wasn't happening.

Thank you !

**Comment by LaurentiuUngur** [comment]
Those are the options that I'm using when a route is calculated:

```swift
            var carOptions = CarOptions()
            carOptions.routeOptions.enableTolls = true
            carOptions.routeOptions.departureTime = Date()
            carOptions.routeOptions.optimizationMode = .fastest
            carOptions.avoidanceOptions.roadFeatures = [.ferry]
```

**Comment by HERE-SDK-Support-Team** [comment]
Hi @LaurentiuUngur, can you provide more details how you calculate the same route twice? Do you hard-code the coordinates in the code? Or do you use the app to tap the same places? Do you clear the map beforehand?

It would be great if you can add the exact steps to reproduce the error. Ideally, with the coordinates of the routes. Thank you.

**Comment by LaurentiuUngur** [comment]
Hi @HERE-SDK-Support-Team. I'm not hardcoding the coordinates in the code. 

The flow is the following: 

1. On screen A I'm choosing origin and destination addresses
2. I'm pushing screen B which has the map where I'm calculating the route from origin to destination (chosen from screen A) and I want to show the calculated route as preview to the user
3. If I go back to screen A and select same origin and destination as before, the route won't be calculated anymore. 

I'll attach the code used in the app that calculates the route (with hardcoded coordinates for testing purpose):
<img width="761" alt="Screenshot 2023-10-13 at 16 17 56" src="https://github.com/heremaps/here-sdk-examples/assets/13499785/7143cc11-2db8-4b3c-9348-50d314bab3a0">

Thank you !

**Comment by LaurentiuUngur** [comment]
@HERE-SDK-Support-Team same happens in newly released SDK v4.16.1

**Comment by HERE-SDK-Support-Team** [comment]
Hi @LaurentiuUngur, in your createRoute() method you also initialize the RoutingEngine. Calculating a route is done asynchronous, that means, you need to keep a reference the engine outside of the scope of the method. Furthermore, when you call the method multiple times, you probably want to reuse an existing instance of the RoutingEngine. Hopefully, this solves your issues. If not, please leave a reply. Thank you!
